actions/ci-beholder-validator/action.yml

```
@@ -0,0 +1,105 @@
+name: "Schema Validator Action"
+description: "Validates schema compatibility with schema registry"
+
+inputs:
+  # general inputs
+  checkout-repo-fetch-depth:
+    description: "number of commits to fetch"
+    required: false
+    default: "0"
+  # image inputs
+  docker-registry:
+    description: "Docker registry to use"
+    required: true
+    default: "aws"
+  image-tag:
+    description: "Image tag to use"
+    required: false
+    default: "latest"
+  # aws inputs
+  aws-role-duration-seconds:
+    description: ""
+    required: false
+    default: "900"
+  aws-region:
+    description: ""
+    required: false
+  aws-account-number:
+    description: ""
+    required: false
+  aws-role-arn:
+    description: ""
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout repo
+      uses: actions/checkout@v4.2.1
+      with:
+        fetch-depth: 0
+    - name: Detect changed files
+      id: changes
+      shell: bash
+      run: |
+        git fetch --prune origin +refs/heads/*:refs/remotes/origin/*
+        # TODO: This logic need to be moved to code
+        changed_files=$(git diff --name-only origin/${{github.event.repository.default_branch }})
+        prefixed_files=""
+        # Loop over each changed file and prepend "./"
+        for file in $changed_files; do
+          prefixed_files="$prefixed_files,./$file"
+        done
+
+        # Remove the leading comma
+        prefixed_files="${prefixed_files#,}"
+
+        echo "CHANGED_FILES=$prefixed_files" >> "$GITHUB_ENV"
+    - name: Start Redpanda with Docker Compose
+      shell: bash
+      run: |
+        docker compose -f "${{ github.action_path }}/docker-compose.yml" up -d redpanda-console
+    - name: Configure aws creds
+      uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502 # v4.0.2
+      if: inputs.docker-registry == 'aws'
+      with:
+        role-to-assume: ${{ inputs.aws-role-arn }}
+        role-duration-seconds: ${{ inputs.aws-role-duration-seconds }}
+        aws-region: ${{ inputs.aws-region }}
+        mask-aws-account-id: true
+    - name: Login to aws ecr
+      id: login-ecr
+      if: inputs.docker-registry == 'aws'
+      uses: aws-actions/amazon-ecr-login@062b18b96a7aff071d4dc91bc00c4c1a7945b076 # v2.0.1
+      with:
+        registries: ${{ inputs.aws-account-number }}
+    - name: Pull schema-validator image from ECR
+      shell: bash
+      run: |
+        docker pull ${{ steps.login-ecr.outputs.registry }}/atlas-beholder-schema-validator:${{ inputs.image-tag }}
+    - name: Validate schemas in master branch
+      shell: bash
+      run: |
+        # Store current branch
+        CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+
+        # Checkout and validate default branch
+        git checkout ${{ github.event.repository.default_branch }}
```

Could this be set as an env variable eg 
```
shell: bash
env:
  DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
```

and then used like this
```suggestion
        git checkout ${{ env.DEFAULT_BRANCH }}
```
Issue [attacker-controllable code ](https://openssf.org/blog/2024/08/12/mitigating-attack-vectors-in-github-workflows/#:~:text=Code%20injection%20on%20GitHub%20Actions,permissions%20granted%20to%20the%20workflow.).

---

Hey @DimitriosNaikopoulos this is mainly done to identify whether a default branch of repo is `main` or `master`. There is no need for the parameterization to be handed to downstream customers. 
Do you see any use cases where this flow is much more beneficial to downstream consumers for validation of schema changes in comparison to the main or master branch?

---

The issue here would be if the `default_branch` was changed to a weird name for code injection. But since the default branch is always controlled by Admins even in public repos this is not as easy to take advantage of (change the branch name to something malicious). @erikburt I think you fixed some issues regarding this on the `.github` repo do you think this needs to be changed as well or because of the nature of the default_branch we are good as is?

---

Will also want to do this with `${{ inputs.image-tag }}`